app/src/main/java/ml/docilealligator/infinityforreddit/NetworkModule.java

```diff
@@ -0,0 +1,212 @@
+package ml.docilealligator.infinityforreddit;
+
+import android.content.SharedPreferences;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import ml.docilealligator.infinityforreddit.utils.APIUtils;
+import okhttp3.ConnectionPool;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.adapter.guava.GuavaCallAdapterFactory;
+import retrofit2.converter.scalars.ScalarsConverterFactory;
+
+@Module(includes = AppModule.class)
+abstract class NetworkModule {
+
+    @Provides
+    @Singleton
+    static OkHttpClient providesBaseOkhttp() {
+        return new OkHttpClient.Builder()
+                .connectTimeout(30, TimeUnit.SECONDS)
+                .readTimeout(30, TimeUnit.SECONDS)
+                .writeTimeout(30, TimeUnit.SECONDS)
+                .build();
+    }
+
+    @Provides
+    @Singleton
+    static Retrofit providesBaseRetrofit(OkHttpClient okHttpClient) {
```

Comment:
Having non-named Retrofit together with named Retrofits is dangerous. If you forget to use `Named` annotation when injecting Retrofit it will work, but not the way you wanted.